src/javascripts/components/cookies-page.js

```diff
@@ -0,0 +1,167 @@
+const saveButtonSelector = '.js-cookies-page-save'
+const successNotificationSelector = '.js-cookies-page-success'
+
+function CookiesPage ($module) {
+  this.$module = $module
+}
+
+CookiesPage.prototype.init = function () {
+  this.$cookiePage = this.$module
+
+  if (!this.$cookiePage) {
+    return
+  }
+
+  this.$cookieFormGroups = this.$cookiePage.querySelectorAll(cookieFormGroupSelector)
+  this.$errorSummary = this.$cookiePage.querySelector(errorSummarySelector)
+  this.$saveButton = this.$cookiePage.querySelector(saveButtonSelector)
+  this.$successNotification = this.$cookiePage.querySelector(successNotificationSelector)
+
+  this.userCookiePreferences = getConsentCookie()
+
+  nodeListForEach(this.$cookieFormGroups, function ($cookieFormGroup) {
+    this.showUserPreference($cookieFormGroup, this.userCookiePreferences)
+    $cookieFormGroup.removeAttribute('hidden')
+  }.bind(this))
+
+  this.$saveButton.addEventListener('click', this.savePreferences.bind(this))
+  this.$saveButton.removeAttribute('hidden')
+}
+
+CookiesPage.prototype.savePreferences = function () {
+  var preferences = {}
+
+  nodeListForEach(this.$cookieFormGroups, function ($cookieFormGroup) {
+    var cookieType = this.getCookieType($cookieFormGroup)
+    var selected = $cookieFormGroup.querySelector(':checked')
+
+    if (selected === null) {
+      this.showErrorMessage($cookieFormGroup)
+      return
+    } else {
+      this.hideErrorMessage($cookieFormGroup)
+    }
+
+    preferences[cookieType] = selected.value === 'yes'
+  }.bind(this))
+
+  // if there are any errors, show error summary and don't submit
+  var $errors = this.$cookiePage.querySelectorAll('.govuk-error-message')
+  if ($errors.length !== 0) {
+    this.showErrorSummary($errors)
+    return
+  }
+
+  // otherwise save preferences to cookie and show success notification
+  setConsentCookie(preferences)
+  this.hideErrorSummary()
+  this.showSuccessNotification()
+}
+
+CookiesPage.prototype.showUserPreference = function ($cookieFormGroup, preferences) {
+  if (!preferences) {
+    return
+  }
+
+  var cookieType = this.getCookieType($cookieFormGroup)
+
+  if (!cookieType || preferences[cookieType] === undefined) {
+    return
+  }
+
+  var preference = preferences[cookieType]
+
+  if (preference === true) {
+    $cookieFormGroup.querySelector('input[value="yes"]').checked = true
+  } else if (preference === false) {
+    $cookieFormGroup.querySelector('input[value="no"]').checked = true
+  }
+}
+
+/* Hide an error message for a cookie option. */
+CookiesPage.prototype.hideErrorMessage = function ($cookieFormGroup) {
+  var errorMessage = $cookieFormGroup.querySelector('.govuk-error-message')
+  if (errorMessage !== null) {
+    errorMessage.parentNode.removeChild(errorMessage)
+  }
+
+  var errorMessageHighlight = $cookieFormGroup.querySelector('.govuk-form-group--error')
+  errorMessageHighlight && errorMessageHighlight.classList.remove('govuk-form-group--error')
```

Comment:
Is the `&&` here just guarding in case nothing matches? Would prefer to be explicit for readability:

```suggestion
  if (errorMessageHighlight) {
    errorMessageHighlight.classList.remove('govuk-form-group--error')
  }
```